packages/babel-plugin-proposal-async-generator-functions/src/index.ts

```diff
@@ -1,66 +1,76 @@
 import { declare } from "@babel/helper-plugin-utils";
 import remapAsyncToGenerator from "@babel/helper-remap-async-to-generator";
 import syntaxAsyncGenerators from "@babel/plugin-syntax-async-generators";
-import { types as t } from "@babel/core";
-import type { PluginPass } from "@babel/core";
 import type { NodePath, Visitor } from "@babel/traverse";
+import { traverse, types as t, type PluginPass } from "@babel/core";
 import rewriteForAwait from "./for-await";
+import environmentVisitor from "@babel/helper-environment-visitor";
 
 export default declare(api => {
   api.assertVersion(7);
 
-  const yieldStarVisitor: Visitor<PluginPass> = {
-    Function(path) {
-      path.skip();
-    },
-
-    YieldExpression({ node }, state) {
-      if (!node.delegate) return;
-      const callee = state.addHelper("asyncGeneratorDelegate");
-      node.argument = t.callExpression(callee, [
-        t.callExpression(state.addHelper("asyncIterator"), [node.argument]),
-        state.addHelper("awaitAsyncGenerator"),
-      ]);
-    },
-  };
+  const yieldStarVisitor = traverse.visitors.merge<PluginPass>([
+    {
+      Function(path) {
+        if (path.isArrowFunctionExpression()) {
+          path.skip();
+        }
+      },
```

Comment:
```suggestion
      ArrowFunctionExpression(path) {
        path.skip();
      },
```
and also in the other places